File: num3.rb
def what_time_is_it(number)
  hours = number / 100
  minutes = number % 100
  (hours >= 0 && hours <= 23) && (minutes >= 0 && minutes <= 59)
end

def yes_or_no(number, n)
  if what_time_is_it(number)
    puts "YES"
  else
    puts "NO"
  end
end

print "n = "
n = gets.chomp.to_i

n.times do
  print "Enter the number: "
  x = gets.chomp.to_i
  yes_or_no(x, n)
end

File: num3.go
package main
import (
	"fmt"
)
func WhatTimeIsIt(number int) bool {
	hours := number / 100
	minutes := number % 100
	return (hours >= 0 && hours <= 23) && (minutes >= 0 && minutes <= 59)
}
func YesOrNo(number int, n int) {
	if WhatTimeIsIt(number) {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}

func main() {
	var n int
	fmt.Print("n = ")
	fmt.Scan(&n)
	for i := 0; i < n; i++ {
		var x int
		fmt.Print("Enter the number: ")
		fmt.Scan(&x)
		YesOrNo(x, n)
	}
}
